test/backoff.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const assert = require('assert');
+
+const Firestore = require('../src');
+const backoff = require('../src/backoff')(Firestore);
+
+const ExponentialBackoff = backoff.ExponentialBackoff;
+const setTimeoutHandler = backoff.setTimeoutHandler;
+
+const nop = () => {};
+
+describe('ExponentialBackoff', function() {
+  const observedDelays = [];
+
+  before(() => {
+    setTimeoutHandler((callback, timeout) => observedDelays.push(timeout));
+  });
+
+  beforeEach(() => {
+    observedDelays.length = 0;
```

Comment:
`observedDelays = []` ?  (just because JavaScript lets you do crazy things doesn't mean you have to :P)

Reply:
I never verified my assumption, but I was worried that the timeout handler captured just the variable at the time and would lose its reference to it when I re-assigned it. But it turns out that that's not how closures work.